File: src/server/Mimirorg.Authentication/Models/Content/MimirorgCompanyCm.cs
﻿using Newtonsoft.Json;

namespace Mimirorg.Authentication.Models.Content
{
    public class MimirorgCompanyCm
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string DisplayName { get; set; }
        public string Description { get; set; }
        public MimirorgUserCm Manager { get; set; }

        [JsonIgnore]
        public string Secret { get; set; }
        public string Domain { get; set; }
        public ICollection<string> Iris { get; set; }
    }
}

File: src/client/src/models/tyle/client/mimirorgCompanyCm.ts
import { MimirorgCompanyCm } from "@mimirorg/typelibrary-types";
import { createEmptyMimirorgUserCm } from "./mimirorgUserCm";

export const createEmptyMimirorgCompanyCm = (): MimirorgCompanyCm => ({
  id: 0,
  name: "",
  displayName: "",
  description: "",
  manager: createEmptyMimirorgUserCm(),
  secret: "",
  domain: "",
  logo: "",
  homePage: "",
  iris: [],
});
